include/systems/equation_systems.h

```diff
@@ -351,28 +353,49 @@ class EquationSystems : public ReferenceCountedObject<EquationSystems>,
                                         std::vector<std::string> & names) const;
 
   /**
-   * Finds system and variable numbers for any variables of \p type
-   * corresponding to the entries in the input 'names' vector.
+   * Finds system and variable numbers for any variables of 'type' or of
+   * 'types' corresponding to the entries in the input 'names' vector.
+   * If 'names' is empty, this returns all variables of the type.
+   * The names of vector variables are decomposed into individual ones
+   * suffixed with their cartesian component, but there will still be
+   * a single pair of system numbers for such vector variables. Thus,
+   * the size of the 'names' vector modified by this function may not
+   * be equal to that of the returned vector of pairs. Nevetheless, both
+   * should be sorted in accordance with ExodusII format, and so the
+   * developer just needs to know to separate dof_indices when
+   * accessing the system solution for vector variables.
+   *
+   * This function is designed to work for either a single type or a
+   * vector of types, but not both. This is because it can't simply be
+   * called a second time with another type because it filters (deletes)
+   * the names of those on the first call which used a different type.
+   * Thus, the 'types' argument is for the case where variables of
+   * multiple types are allowed to pass through.
```

Comment:
Remind me - what was the expected behavior if we only pass in nullptr - was it "return all variable numbers for variables of every type"?  My fault for not documenting it initially, but we should get it squared away for the future, especially if we're making this function more complicated.

Reply:
Yes, all variables listed in `names` no matter there type, would be returned with their numbers. Similarly, if the `names` list is empty, all variables of specified `type` OR `types` no matter what there name is, would be returned.

Thus, if `names` is empty and neither `type` nor `types` is specified, every single variable in the system gets returned.

Reply:
This was how it worked before, the only difference is not its possible to provide a list of types so that this function need only be called once. Awhile ago we discussed whether we should make `type` a vector by default, but we were concerned we'd be break people's code if we did that.